contracts/proxy-lib/src/lib.rs

```diff
@@ -0,0 +1,194 @@
+        let proposal = proposal
+            .parse(|i| *i.signer_id)
+            .expect("failed to parse input");
+
+        let singer_id = &proposal.signer_id;
+
+        let num_proposals = self.num_proposals_pk.get(singer_id).unwrap_or(&0) + 1;
+        assert!(
+            num_proposals <= self.active_proposals_limit,
+            "Account has too many active proposals. Confirm or delete some."
+        );
+        self.num_proposals_pk
+            .insert(singer_id.clone(), num_proposals);
+
+        self.proposals.insert(self.proposal_nonce, proposal.clone());
+        self.approvals.insert(self.proposal_nonce, HashSet::new());
+        self.proposal_nonce += 1;
+        self.proposal_nonce - 1
+    }
+
+    pub fn create_and_approve_proposal(&mut self, proposal: Signed<ProposalWithSigner>) -> Promise {
+        let proposal_id = self.create_proposal(proposal.clone());
+        let proposal = proposal
+            .parse(|i| *i.signer_id)
+            .expect("failed to parse input");
+        log!("Starting approval...");
+        return self.approve_by_member(proposal.signer_id, proposal_id);
+    }
+
+    pub fn approve(&mut self, request: Signed<ConfirmationRequestWithSigner>) -> Promise {
+        let request = request
+            .parse(|i| *i.signer_id)
+            .expect("failed to parse input");
+        return self.approve_by_member(request.signer_id, request.proposal_id);
+    }
+
+    fn internal_confirm(&mut self, request_id: ProposalId, signer_id: Repr<SignerId>) -> () {
+        let confirmations = self.approvals.get_mut(&request_id).unwrap();
+        assert!(
+            !confirmations.contains(&signer_id),
+            "Already confirmed this request with this key"
+        );
+        confirmations.insert(signer_id);
+
+        // Check if the number of confirmations is enough to execute the request
+        // If so, execute the request
+        // If not, do nothing
+    }
+
+    fn approve_by_member(&self, identity: Repr<SignerId>, request_id: ProposalId) -> Promise {
+        log!("Starting fetch_members...");
+        config_contract::ext(self.context_config_account_id.clone())
+            .with_static_gas(Gas::from_tgas(5))
+            .has_member(self.context_id, identity)
+            .then(
+                Self::ext(env::current_account_id()).internal_process_members(identity, request_id),
+            )
+    }
+
+    pub fn requests(&self, offset: usize, length: usize) -> Vec<(&u32, &ProposalWithSigner)> {
+        let mut requests = Vec::with_capacity(length);
+        for request in self.proposals.iter().skip(offset).take(length) {
+            requests.push(request);
+        }
+        requests
+    }
+
+    pub fn get_confirmations_count(&self, proposal_id: ProposalId) -> ProposalWithApprovals {
+        let size = self
+            .approvals
+            .get(&proposal_id)
+            .unwrap_or(&HashSet::new())
+            .len();
```

Comment:
shouldn't we return an option here if the proposal doesn't exist?

```suggestion
    pub fn get_confirmations_count(&self, proposal_id: ProposalId) -> Option<ProposalWithApprovals> {
        let size = self
            .approvals
            .get(&proposal_id)?
            .len();
```

Reply:
Fair